app/javascript/components/search/genes/GeneKeyword.js

```diff
@@ -0,0 +1,75 @@
+import React, { useContext, useState, useRef } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faTimes, faSearch } from '@fortawesome/free-solid-svg-icons'
+import _clone from 'lodash/clone'
+import Button from 'react-bootstrap/lib/Button'
+import Modal from 'react-bootstrap/lib/Modal'
+
+import { GeneSearchContext } from 'providers/GeneSearchProvider'
+import { StudySearchContext } from 'providers/StudySearchProvider'
+
+/** renders the gene text input
+  * This is split into its own component both for modularity, and also because
+  * having it inlined in GeneSearchView led to a mysterious infinite-repaint bug in StudyResults
```

Comment:
```suggestion
  * having it inlined in GeneSearchView led to a mysterious infinite-repaint bug in StudyResults
```

Interesting!  Glad something of a cause has been found.